silx/gui/icons.py

```diff
@@ -29,11 +29,131 @@
 
 __authors__ = ["T. Vincent"]
 __license__ = "MIT"
-__date__ = "12/05/2016"
+__date__ = "01/09/2016"
 
 
+import logging
+import weakref
 from . import qt
 from ..resources import resource_filename
+from ..utils import weakref as silxweakref
+
+
+_logger = logging.getLogger(__name__)
+"""Module logger"""
+
+
+_cached_icons = weakref.WeakValueDictionary()
+"""Cache loaded icons in a weak structure"""
+
+
+_supported_formats = None
+"""Order of file format extension to check"""
+
+
+_process_working = None
+"""Cache an AnimatedIcon for working process"""
+
+
+def get_process_working():
+    """Returns a cached version of the AnimatedIcon pro cess-working.
```

Comment:
API style consistency: get_process_working -> getProcessWorking
The name is confusing to me, I'd call it something like getWaitIcon or getBusyAnimatedIcon

Typo: process-working